connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClientCache.scala

```diff
@@ -66,8 +67,9 @@ object KustoClientCache {
       )
     }
 
-    engineKcsb.setClientVersionForTracing(KCONST.ClientName)
-    ingestKcsb.setClientVersionForTracing(KCONST.ClientName)
+    val sparkVersion = SparkSession.builder().getOrCreate().version
+    engineKcsb.setConnectorDetails(KCONST.ClientName, KustoDataSourceUtils.Version ,null, null, false, null,  Pair.of("spark.version", sparkVersion))
```

Comment:
we can actually get the application name 
we have it in KustoWriter - maybe you can get it here the same way as well
data.sparkSession.sparkContext.appName